contracts/multisig-prover/src/encoding/evm.rs

```diff
@@ -44,34 +46,69 @@ impl TryFrom<Message> for Command {
     }
 }
 
-impl TryFrom<Signer> for Operator {
+impl TryFrom<WorkerSet> for Command {
+    type Error = ContractError;
+    fn try_from(worker_set: WorkerSet) -> Result<Self, Self::Error> {
+        let params = transfer_operatorship_params(&worker_set)?;
+        Ok(Command {
+            ty: CommandType::TransferOperatorship,
+            params,
+            id: worker_set.hash(),
+        })
+    }
+}
+
+impl TryFrom<(Signer, Option<Signature>)> for Operator {
     type Error = ContractError;
 
-    fn try_from(signer: Signer) -> Result<Self, Self::Error> {
+    fn try_from((signer, sig): (Signer, Option<Signature>)) -> Result<Self, Self::Error> {
         Ok(Self {
             address: evm_address((&signer.pub_key).into())?,
             weight: signer.weight,
-            signature: signer.signature.map(|Signature::ECDSA(sig)| sig),
+            signature: sig.map(|Signature::ECDSA(sig)| sig),
         })
     }
 }
 
+pub fn make_worker_set(
+    snapshot: Snapshot,
+    pub_keys: Vec<multisig::types::PublicKey>,
+) -> Result<Operators, ContractError> {
+    let addresses = pub_keys
+        .iter()
+        .map(|pub_key| evm_address(pub_key.into()))
+        .collect::<Result<Vec<HexBinary>, _>>()?;
+    let zipped: Vec<(HexBinary, Uint256)> = addresses
+        .into_iter()
+        .zip(snapshot.participants.values().map(|p| p.weight.into()))
+        .collect();
+    Ok(Operators {
+        weights: zipped,
+        threshold: snapshot.quorum.into(),
+    })
+}
+
 impl CommandBatch {
     pub fn new(
         messages: Vec<Message>,
         destination_chain_id: Uint256,
+        new_worker_set: Option<WorkerSet>,
```

Comment:
I'd prefer something like following instead of having everything possible in the `.new` method.
```
let command_batch_builder = CommandBatchBuilder::default();

command_batch_builder.add_message( ... );
command_batch_builder.add_new_worker_set( ... );

let command_batch = command_batch_builder.build();
```

Reply:
We are likely going to have more types of commands, and keep adding things to `new` is confusing to me.